data/service/api/v1/datasets_update.go

```diff
@@ -84,16 +84,20 @@ func DataSetsUpdate(dataServiceContext dataService.Context) {
 	}
 
 	if update.State != nil && *update.State == "closed" {
-		deduplicator, getErr := dataServiceContext.DataDeduplicatorFactory().Get(dataSet)
-		if getErr != nil {
-			dataServiceContext.RespondWithInternalServerFailure("Unable to get deduplicator", getErr)
-			return
-		} else if deduplicator == nil {
-			dataServiceContext.RespondWithInternalServerFailure("Deduplicator not found")
-			return
-		} else if err = deduplicator.Close(ctx, dataServiceContext.DataRepository(), dataSet); err != nil {
-			dataServiceContext.RespondWithInternalServerFailure("Unable to close", err)
-			return
+		// Allow closing datasets without deduplicator (i.e. those created by jellyfish)
+		// so we can mark summaries as outdated.
+		if dataSet != nil && dataSet.HasDeduplicatorName() {
```

Comment:
If I'm understanding things correctly, I don't think this will cause issues with the current Jellyfish migration/deprecation plan. The plan does not currently include any updates to the data set (just data). Thus, a Jellyfish data set will not have a deduplicator name and so this code will not be executed.

If, however, we do need to update the data set with the deduplicator name, then this will cause a couple of potential issues:

1. The Jellyfish data set and data may include some fields normally only present in Platform data (written by `DataRepository.UpdateDataSet` and `DataRepository.ActiveDataSetData`). We'd need to do a survey of exactly what fields are added by Jellyfish or Platform and compare the two. Even if there are field differences they may not cause any real issue, just inconsistencies.
2. The Platform code that performs data duplication will execute on Jellyfish data (where other Jellyfish data is duplicated based upon this Jellyfish data using the Platform deduplication mechanism). I don't *think* this will cause an issue, though, since Jellyfish will have already deduplicated the data (by dropping any new duplicates) prior to closing of the Jellyfish data set. Thus, I *think* that this data deduplication step would end up just being a NOP.

In either case, we should verify this definitely won't cause issues and add it as one or more test cases, but at first glance there doesn't appear to be any (or if there are some they would be minor).

cc: @jh-bate